### ISSUE TITLE
Inconsistent output with pre-trained model

### ISSUE THREAD

**Issue by strin** [facebookresearch/InferSent#48]
Using pre-trained model `infersent.allnli.pickle`, we seem to get inconsistent results when place the sentence in different batches. As a concrete example, let the sentence be `hello, world.`, then

```
In [44]: infersent.encode(['hello, world.'], tokenize=True, verbose=True)
array([[ 0.14178185, -0.00886203, -0.03044512, ..., -0.03926747,
        -0.03814263, -0.00360173]], dtype=float32)
```

But 

```
In [45]: encoder.infersent.encode(['hello, world.', 'a longer sentence is placed in the
    ...: batch'], tokenize=True, verbose=True)
array([[ 0.14178185,  0.        ,  0.        , ...,  0.        ,
         0.        ,  0.        ],
       [ 0.0756584 ,  0.08044495,  0.05892764, ...,  0.03711559,
         0.00114529,  0.02401241]], dtype=float32)
```

Looks like many dimensions of the embedding for `hello, world.` become zero. Checking the dropout ratio, I am sure dropout is not used (`dpout_model = 0`).

**Comment by zarzen** [comment]
Same issue, did you find the answer?

**Comment by lsgh** [comment]
I have the same strange issue. Could anyone comment on this?

**Comment by zarzen** [comment]
@lsgh try the modification made by ihsgnef, here is his repo https://github.com/ihsgnef/InferSent-1 . I have tested, it works on my side.

**Comment by aconneau** [comment]
Hi,
I uploaded a second model that was trained without this feature, which I called "max_pool" in 14b7ebb01d9186e9130330deb720367ce67bb9f4 .
With the infersent2 model, by specifying "version": 2 in the parameters, you shouldn't have this problem anymore. Note that the loading of the InferSent model changed. Please take a look at the README and the example here: https://github.com/facebookresearch/InferSent/blob/master/encoder/demo.ipynb
Thanks,
Alexis

**Comment by rachit-hotstar** [comment]
@aconneau - The above embeddings seem to be different on the 8th decimal place.
`[ 0.00592252, 0.13414043, -0.17540528, ..., 0.03551164, -0.01075476, 0.05849883]`
`[ 0.00592252, 0.13414048, -0.17540526, ..., 0.03551164, -0.01075473, 0.05849886]`

Similar to this I am noticing that the vectors are different if I produce them on a different machine.
For eg: on my machine I am getting: 
`[ 0.00746889, -0.1127886 ,  0.0492736 , ...,  0.00274666, 0.03849312, -0.0138564 ]` with 2 sentences and 
`[ 0.00746889, -0.1127886 ,  0.04927361, ...,  0.00274666, 0.03849312, -0.0138564 ]` with just one sententece.

The embeddings seem to differ on different machines. Is that a known issue? And is there a way to solve for this? By setting a random seed somewhere?
